ui.frontend/src/view/FormFieldBase.js

```diff
@@ -351,31 +351,72 @@ class FormFieldBase extends FormField {
         }
     }
 
+    /**
+     * Updates the HTML state based on the required state of the field.
+     * @param {boolean} required - The required state.
+     * @param {Object} state - The state object.
+     */
+    updateRequired(required, state) {
+        if (this.widget) {
+            this.toggle(required, "required");
+            if (required === true) {
+                this.widget.setAttribute("required", "required");
+            } else {
+                this.widget.removeAttribute("required");
```

Comment:
AFAIR this should take care of accessibility, @pankaj-parashar do let us know if something else is needed too?